mep3_localization/mep3_localization/robot_detection.py

```diff
@@ -0,0 +1,211 @@
+    orientations and sends TF of Aruco tags.
+    """
+
+    def __init__(self):
+        super().__init__('image_subscriber')
+        self.image_subscription = self.create_subscription(
+            Image, '/cam/cam_central/RasPi0', self.image_listener_callback, 10)
+        self.image_subscription
+        self.camera_info_subscription = self.create_subscription(
+            CameraInfo, '/cam/cam_central/RasPi0/camera_info',
+            self.camera_info_listener_callback, 10)
+        self.camera_info_subscription
+        self._tf_publisher = TransformBroadcaster(self)
+        self.static_tf_listener('map', 'camera_static')
+
+        self.dict = aruco.Dictionary_get(aruco.DICT_4X4_100)
+        self.params = aruco.DetectorParameters_create()
+        # Side length of the ArUco marker in meters
+        self.aruco_robot_length = 0.07
+        self.aruco_42_length = 0.1
+
+    def image_listener_callback(self, data):
+        # self.get_logger().info('Receiving video frame')
+        br = CvBridge()
+        current_frame = br.imgmsg_to_cv2(data, 'bgr8')
+        rvecs, tvecs, ids = self.get_aruco_pose(current_frame)
+        self.send_pose_tf2(rvecs, tvecs, ids)
+
+    def static_tf_listener(self, parent, child):
+        # self.camera_translation = [-0.141, 1.417, 1.184]
+        # self.camera_rotation = R.from_quat([-0.005, 0.962, -0.272, 0.008])
+        self.tf_buffer = Buffer()
+        self.tf_listener = TransformListener(self.tf_buffer, self)
+        trans = None
+        while not trans:
+            try:
+                trans = self.tf_buffer.lookup_transform(
+                    child, parent, rclpy.time.Time())
+            except Exception:  # noqa: E501
+                # logging.exception("StaticTfListener"):
+                pass
+            rclpy.spin_once(self, timeout_sec=0.1)
+
+        self.camera_translation = [
+            trans.transform.translation.x, trans.transform.translation.y,
+            trans.transform.translation.z
+        ]
+        self.camera_rotation = R.from_quat([
+            trans.transform.rotation.x, trans.transform.rotation.y,
+            trans.transform.rotation.z, trans.transform.rotation.w
+        ])
+
+    def camera_info_listener_callback(self, data):
+        # self.get_logger().info('Receiving camera info frame')
+        # self.cameraMatrix = np.array([[570.34, 0, 1920 / 2],
+        #                               [0, 570.34, 1080 / 2], [0, 0, 1]])
+        # self.distCoeffs = np.zeros((5, 1))
+        self.cameraMatrix = np.array([[data.k[0], data.k[1], data.k[2]],
+                                      [data.k[3], data.k[4], data.k[5]],
+                                      [data.k[6], data.k[7], data.k[8]]])
+        self.distCoeffs = np.array([[i] for i in data.d])
+
+    def send_pose_tf2(self, rvecs, tvecs, ids):
+        if ids is not None:
+            for i in range(len(rvecs)):
+                if ids[i] < 10 or ids[i] == 42:
```

Comment:
Mozda negde definisati ove konstante 10 i 42, pa koristiti po imenu?
Ovako izgledaju kao proizvoljni brojevi pa bi se u buducnosti, ako se menja, moralo traziti u dokumentaciji zasto je to tako napisano...